src/components/Forms/Forms.js

```diff
@@ -1,26 +1,178 @@
+const isPageWithoutArray = (CurrentPage) => {
+  return (
+    CurrentPage === "About" ||
+    CurrentPage === "Skills" ||
+    CurrentPage === "Hobbies" ||
+    CurrentPage === "Contact"
+  );
+};
+
+const Forms = ({
+  CurrentPage,
+  SetSection,
+  Resume,
+  SetSwitch,
+  ImageHandler,
+}) => {
+  // This function is used to add new Item to the List ----------------------------------------
+  const addItem = (CurrPage) => {
+    if (CurrPage === "Projects") {
+      addItemToProject();
+    } else if (CurrPage === "Experiences") {
+      addItemToExperiences();
+    } else if (CurrPage === "Awards") {
+      addItemToAwards();
+    } else if (CurrPage === "Educations") {
+      addItemToEducations();
+    }
+  };
+  const addItemToProject = () => {
+    var newList = [...Resume["Projects"].List, dummyProject];
+    const newItemIndex = newList.length - 1;
+    SetSection("Projects", {
+      Current: newItemIndex,
+      List: newList,
+    });
+  };
+  const addItemToExperiences = () => {
+    var newList = [...Resume["Experiences"].List, dummyExperience];
+    const newItemIndex = newList.length - 1;
+    SetSection("Experiences", {
+      Current: newItemIndex,
+      List: newList,
+    });
+  };
+  const addItemToAwards = () => {
+    var newList = [...Resume["Awards"].List, dummyAward];
+    const newItemIndex = newList.length - 1;
+    SetSection("Awards", {
+      Current: newItemIndex,
+      List: newList,
+    });
+  };
+  const addItemToEducations = () => {
+    var newList = [...Resume["Educations"].List, dummyEducation];
+    const newItemIndex = newList.length - 1;
+    SetSection("Educations", {
+      Current: newItemIndex,
+      List: newList,
+    });
+  };
+  // ----------------------------------------------------------------------------------------------
+
   return (
     <div className="rezume-forms">
-      {Resume[CurrentPage] &&
-        Object.keys(Resume[CurrentPage]).map((fe, key) => (
-          <FormElement
-            key={key}
-            Type={fe === "Summary" ? "TextArea" : fe === "ProfilePicture" ? "Switch" : "Text"}
-            Label={fe}
-            Value={Resume[CurrentPage][fe]}
-            onChange={e =>
-              SetSection(CurrentPage, {
-                ...Resume[CurrentPage],
-                [fe]: e.target.value
-              })
-            }
-            onSwitch={() => SetSwitch()}
-            ImageHandler={e => ImageHandler(e)}
-          />
-        ))}
+      {/* This button is to add new items in the array. So we only need to display it for the pages that contain an array / List */}
+      {isPageWithoutArray(CurrentPage) ? (
+        <></>
```

Comment:
What's this doing?

Reply:
We have two types of sections in this project one that does not have a list of elements like About, Skills, etc, and others with a list of elements like Experience, Awards. In the one without a list, we simply map through the attributes of the object. But in terms of the list items, we need to map through a single item in the "list" (list[index].map) so we need to have two different types of rendering for the form on the left. This isPageWithoutArray() helps decide whether the current page (Experience or Award or about etc) has a list/array or not. I have defined this function above. 

Alternatively, we could have used a long conditional statement like (currentPage === "Awards" || .....) but to make it more readable I made it a function.

I hope this made the code more readable and not messy 😅 .